cmd/notation/registry.go

```diff
@@ -14,39 +15,52 @@ import (
 	"oras.land/oras-go/v2/registry/remote/auth"
 )
 
-func getSignatureRepository(opts *SecureFlagOpts, reference string) (notationregistry.Repository, error) {
+func getSignatureRepositoryClient(opts *SecureFlagOpts, reference string) (*remote.Repository, error) {
 	ref, err := registry.ParseReference(reference)
 	if err != nil {
 		return nil, err
 	}
 	return getRepositoryClient(opts, ref)
 }
 
-func getRegistryClient(opts *SecureFlagOpts, serverAddress string) (*remote.Registry, error) {
-	reg, err := remote.NewRegistry(serverAddress)
+func getRepositoryClient(opts *SecureFlagOpts, ref registry.Reference) (*remote.Repository, error) {
+	authClient, plainHTTP, err := getAuthClient(opts, ref)
 	if err != nil {
 		return nil, err
 	}
+	return &remote.Repository{
+		Client:    authClient,
+		Reference: ref,
+		PlainHTTP: plainHTTP,
+	}, nil
+}
 
-	reg.Client, reg.PlainHTTP, err = getAuthClient(opts, reg.Reference)
-	if err != nil {
-		return nil, err
+func setHttpDebugLog(repo *remote.Repository, debug bool) {
+	if !debug {
+		return
+	}
+	if authClient, ok := repo.Client.(*auth.Client); ok {
+		if authClient.Client == nil {
+			authClient.Client = http.DefaultClient
+		}
+		if authClient.Client.Transport == nil {
+			authClient.Client.Transport = http.DefaultTransport
+		}
+		authClient.Client.Transport = trace.NewTransport(authClient.Client.Transport)
 	}
-	return reg, nil
 }
 
-func getRepositoryClient(opts *SecureFlagOpts, ref registry.Reference) (notationregistry.Repository, error) {
-	authClient, plainHTTP, err := getAuthClient(opts, ref)
+func getRegistryClient(opts *SecureFlagOpts, serverAddress string) (*remote.Registry, error) {
+	reg, err := remote.NewRegistry(serverAddress)
 	if err != nil {
 		return nil, err
 	}
-	repo := &remote.Repository{
-		Client:    authClient,
-		Reference: ref,
-		PlainHTTP: plainHTTP,
-	}
 
-	return notationregistry.NewRepository(repo), nil
+	reg.Client, reg.PlainHTTP, err = getAuthClient(opts, reg.Reference)
+	if err != nil {
+		return nil, err
+	}
+	return reg, nil
 }
 
 func getAuthClient(opts *SecureFlagOpts, ref registry.Reference) (*auth.Client, bool, error) {
```

Comment:
Indeed, we need to pass a `debug bool` to enable tracing for http requests.

Reply:
Otherwise, you will miss it for future development like `notation list`.

Reply:
Debugging is optional and user shouldn't require to pass a parameter for optional feature.

We can add unit tests to make sure that we are emitting output.

Reply:
If we add a debug flag here, all the callers of this function also need to receive a `debug` argument, which looks not quite right.

Reply:
Update: added HttpDebugLog Option to be optional field.

Reply:
Update: removed Options, and added a context variable to trace the debug status